### PR TITLE
Don't push checkpoints to hub in `no_trainer` scripts

### PR DESCRIPTION
**Pull request by muellerzr**
# Don't push checkpoints to the Hub in `no_trainer` scripts

## What does this add?

- Creates a `gitignore` file in the base folder if `push_to_hub` was passed and one does not exist
- During each call to `save_state`, if `push_to_hub` was passed then then directory is added to the .gitignore

## Why is it needed?

Users shouldn't expect to have all of their checkpoints sent to both the Hub and saved locally, it should just be locally until they are ready for the final save

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
I think we can just set the gitignore with a general pattern at its creation.

File: examples/pytorch/language-modeling/run_clm_no_trainer.py [hunk 1]
```diff
@@ -258,6 +258,9 @@ def main():
             else:
                 repo_name = args.hub_model_id
             repo = Repository(args.output_dir, clone_from=repo_name)
+            gitignore_path = os.path.join(args.output_dir, ".gitignore")
+            if not os.path.exists(gitignore_path):
+                open(gitignore_path, "a").close()  # Creates a .gitignore
```
[the remark this answers is not in the document]
Comment (muellerzr) [reply examples/pytorch/language-modeling/run_clm_no_trainer.py:263]: Ah, yeah that would be a bit more simplistic. Thanks!

[the remark this answers is not in the document]
Comment (sgugger) [reply examples/pytorch/language-modeling/run_clm_no_trainer.py:263]: Nice!

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

**Pull request merged**